### PR TITLE
Integrate new Datumaro API for tests

### LINKED ISSUE #5717: Update OTX integration tests to use new Datumaro datasets and API
Several integration tests in OTX still rely on the old Datumaro dataset interface to load the test data. The goal of this ticket is to modernize these tests to use the new DM format and API, to make them not only faster but also more aligned with a typical usage of the library.

In scope of this ticket, replace `test_engine_api.py` + `test_geti_interaction.py` with a single `test_engine.py` that:
- takes a tiny dataset, model architecture and a minimal training configuration as input
- is parametrized to test one model architecture per task
- tests the typical OTX workflow with the engine:
  - instantiates a datamodule and model
  - instantiates the engine
  - runs training for 1-2 epochs
  - evaluates the model (test)
  - gets a prediction with the model
  - exports the model to ONNX and gets a prediction with it
  - exports the model to OV and gets a prediction with it
  - quantizes the model to INT8 and gets a prediction with it

This test is meant to:
- execute relatively fast, so that it can be run on PR merge
- block PRs that fundamentally break the core e2e workflow of OTX

Testing more scenarios and configurations will be instead covered by separate e2e tests.

### PR DESCRIPTION
**Pull request by AlbertvanHouten**
This PR refactors the data pipeline to fully adopt the new `datumaro.experimental` API and simplifies dataset handling throughout the OTX codebase. The main changes include updating testing datasets to the new format (including adding download scripts to create them) and completely removing legacy datumaro logic. A lot of dead code has also been removed if it was referencing the legacy datumaro logic.

All new datasets are at most 1.5mb each. 

### Checklist

- [ ] The PR title and description are clear and descriptive
- [ ] I have manually tested the changes
- [ ] All changes are covered by automated tests
- [ ] All related issues are linked to this PR (if applicable)
- [ ] Documentation has been updated (if applicable)

### CODE REVIEW

File: library/src/otx/data/dataset/keypoint_detection.py [hunk 1]
```diff
@@ -70,9 +70,9 @@ def __init__(
             to_tv_image=to_tv_image,
             data_format=data_format,
         )
-        labels = dm_subset.schema.attributes["label"].categories.labels
+        labels = dm_subset.schema.attributes["keypoints"].categories.labels
```
Comment (leoll2) [inline library/src/otx/data/dataset/keypoint_detection.py:69]: Is this a bugfix related to https://github.com/open-edge-platform/datumaro/pull/2070?

File: library/src/otx/backend/native/models/segmentation/litehrnet.py [hunk 2]
```diff
@@ -104,7 +104,10 @@ def _exporter(self) -> OTXModelExporter:
             pad_value=0,
             swap_rgb=False,
             via_onnx=False,
-            onnx_export_configuration={"operator_export_type": OperatorExportTypes.ONNX_ATEN_FALLBACK},
+            onnx_export_configuration={
+                "operator_export_type": OperatorExportTypes.ONNX_ATEN_FALLBACK,
+                "dynamo": False,
```
Comment (leoll2) [inline library/src/otx/backend/native/models/segmentation/litehrnet.py:109]: Does Dynamo tracing fail for this model?

File: application/backend/uv.lock [hunk 3]
```diff
@@ -87,13 +86,13 @@ name = "aiohttp"
...
 source = { registry = "https://pypi.org/simple" }
 dependencies = [
-    { name = "aiohappyeyeballs" },
-    { name = "aiosignal" },
-    { name = "attrs" },
-    { name = "frozenlist" },
-    { name = "multidict" },
-    { name = "propcache" },
-    { name = "yarl" },
+    { name = "aiohappyeyeballs", marker = "(extra == 'extra-3-otx-cpu' and extra == 'extra-3-otx-cuda') or (extra == 'extra-3-otx-cpu' and extra == 'extra-3-otx-xpu') or (extra == 'extra-3-otx-cpu' and extra == 'extra-9-geti-tune-cpu') or (extra == 'extra-3-otx-cuda' and extra == 'extra-3-otx-xpu') or (extra == 'extra-3-otx-cuda' and extra == 'extra-9-geti-tune-cuda') or (extra == 'extra-3-otx-xpu' and extra == 'extra-9-geti-tune-xpu') or (extra == 'extra-9-geti-tune-cpu' and extra == 'extra-9-geti-tune-cuda') or (extra == 'extra-9-geti-tune-cpu' and extra == 'extra-9-geti-tune-xpu') or (extra == 'extra-9-geti-tune-cuda' and extra == 'extra-9-geti-tune-xpu')" },
```
Comment (leoll2) [inline application/backend/uv.lock:89]: FYI I noticed that `rm uv.lock && uv lock` produces a cleaner lock file than just `uv lock`, with less of this boilerplate markers.

File: library/src/otx/data/entity/sample.py [hunk 4]
```diff
@@ -106,6 +106,7 @@ def img_info(self, value: ImageInfo) -> None:
 
 
 @register_pytree_node
+@register_sample
 class ClassificationSample(OTXSample):
```
Comment (leoll2) [inline library/src/otx/data/entity/sample.py:110]: If you register this sample for dataset import, will it not interfere with the [`ClassificationSample`](https://github.com/open-edge-platform/training_extensions/blob/f31fd92495613d50934caaf8f634cdb15a23fe59/application/backend/app/datumaro_converter/domain/samples.py#L44) of the Geti backend?

File: library/src/otx/data/module.py [hunk 5]
```diff
@@ -101,21 +109,10 @@ def __init__(
...
-            self.task == OTXTaskType.KEYPOINT_DETECTION and self.data_format == "arrow"
-        ):
-            dataset = pre_filtering(
-                dataset,
-                self.data_format,
-                self.unannotated_items_ratio,
-                self.task,
-                ignore_index=self.ignore_index if self.task == "SEMANTIC_SEGMENTATION" else None,
-            )
+            self.data_format = dataset.format if hasattr(dataset, "format") else "datumaro"
```
Comment (leoll2) [inline library/src/otx/data/module.py]: What is `OTXDataModule.data_format` needed for?

File: library/src/otx/types/label.py [hunk 6]
```diff
@@ -36,11 +45,7 @@ def __eq__(self, other: object) -> bool:
         """Compare two LabelInfo objects, normalizing list/tuple differences."""
         if not isinstance(other, LabelInfo):
             return NotImplemented
-        return (
-            list(self.label_names) == list(other.label_names)
-            and list(self.label_ids) == list(other.label_ids)
-            and [list(g) for g in self.label_groups] == [list(g) for g in other.label_groups]
-        )
+        return _normalize(asdict(self)) == _normalize(asdict(other))
```
Comment (leoll2) [inline library/src/otx/types/label.py]: Two questions here:
1. What's the usecase for comparing `LabelInfo` objects?
2. All attributes are typed as `list`, so the normalization shouldn't be needed unless some code forcibly initializes these attributes with iterables of the wrong type

File: library/pyproject.toml [hunk 7]
```diff
@@ -73,6 +73,8 @@ dev = [
     "pytest-forked",
     "dill",
     "tifffile",
+    "py-cpuinfo",
+    "mlflow",
```
Comment (leoll2) [inline library/pyproject.toml:77]: Where are these new dependencies used?

**Review by leoll2 — COMMENTED** [review]
Massive work, thank you. I haven't finished reviewing everything yet, but I can already share some some comments.

File: library/src/otx/data/dataset/keypoint_detection.py (hunk 1 above)
Comment (AlbertvanHouten) [reply library/src/otx/data/dataset/keypoint_detection.py:69]: Yes, I believe the original keypoint testing dataset had labels that consisted of just 1 keypoint and that is why it previously worked. The new coco testing dataset has 17 keypoints for the Person label and the model needs to know about these 17 keypoints instead of the single person label.

File: library/src/otx/backend/native/models/segmentation/litehrnet.py (hunk 2 above)
Comment (AlbertvanHouten) [reply library/src/otx/backend/native/models/segmentation/litehrnet.py:109]: OperatorExportTypes.ONNX_ATEN_FALLBACK doesn't work anymore with the current pytorch version in OTX ( I guess this model was not tested previously). Disabling dynamo falls back to legacy logic.

File: library/src/otx/data/entity/sample.py (hunk 4 above)
Comment (AlbertvanHouten) [reply library/src/otx/data/entity/sample.py:110]: No, class names are irrelevant for sample matching. It will look at the schema of the sample and exact field configuration.

File: library/pyproject.toml (hunk 7 above)
Comment (AlbertvanHouten) [reply library/pyproject.toml:77]: Both were already in use in the regression tests.

File: library/src/otx/data/module.py (hunk 5 above)
Comment (AlbertvanHouten) [reply library/src/otx/data/module.py]: Its no longer needed, good catch. I have removed it entirely.

File: application/backend/uv.lock (hunk 3 above)
Comment (AlbertvanHouten) [reply application/backend/uv.lock:89]: Seems to not work for me. I redid it this morning in https://github.com/open-edge-platform/training_extensions/pull/5740/commits/653144451ca3b3fef3bdf9a8066d40797434b89d but still the same style lock file was generated.

File: library/src/otx/types/label.py (hunk 6 above)
Comment (AlbertvanHouten) [reply library/src/otx/types/label.py]: Testing dataset is comparing its labels against the training dataset labels and they seem to get initialized through different paths. I have now fixed the label info to receive a list from the dataset classes.

File: application/backend/uv.lock (hunk 3 above)
Comment (leoll2) [reply application/backend/uv.lock:89]: No problem, thanks for trying

**Review by leoll2 — DISMISSED** [review]
LGTM. Before merging, please test the changes in Geti by training a model for each task.

**Review by leoll2 — APPROVED** [review]
